security/android_vbmeta.ksy

```diff
@@ -0,0 +1,192 @@
+      - id: descriptors
+        type: position
+        doc: Position of descriptor data in "Auxiliary data" block.
+      - id: rollback_index
+        type: u8
+        doc: |
+          The rollback index which can be used to prevent rollback to
+          older versions.
+      - id: flags
+        type: u4
+        doc: |
+          Flags from the AvbVBMetaImageFlags enumeration. This must be
+          set to zero if the vbmeta image is not a top-level image.
+      - id: reserved0
+        contents: [0, 0, 0, 0]
+        doc: |
+          Reserved to ensure |release_string| start on a 16-byte boundary.
+          Must be set to zeroes.
+      - id: release_string
+        type: strz
+        size: 48
+        encoding: ASCII
+        doc: |
+          The release string from avbtool, e.g. "avbtool 1.0.0" or
+          "avbtool 1.0.0 xyz_board Git-234abde89". Is guaranteed to be NUL
+          terminated. Applications must not make assumptions about how this
+          string is formatted.
+      - id: reserved
+        type: u1
+        repeat: expr
+        repeat-expr: 80
+        doc: |
+          Padding to ensure struct is size AVB_VBMETA_IMAGE_HEADER_SIZE
+          bytes. This must be set to zeroes.
+    instances:
+      hashtree_disabled:
+        value: flags & (1 << 0)
+      verification_disabled:
+        value: flags & (1 << 1)
+  vbmeta_public_key_header:
+    doc: The header for a serialized RSA public key
+    seq:
+      - id: key_num_bits
+        type: u4
+        doc: The size of the key in bits
+      - id: n0inv
+        type: u4
+        doc: Precomputed value for optimization of verification.
+  vbmeta_descriptor:
+    seq:
+      - id: tag
+        type: u8
+        enum: descriptor_types
+      - id: num_bytes_following
+        type: u8
+      - id: data
+        size: num_bytes_following
+  vbmeta_descriptors:
+    seq:
+      - id: descriptors
+        type: vbmeta_descriptor
+        repeat: eos
+  vbmeta_authentication_data:
+    instances:
+      hash_body:
+        pos: _root.header.hash.offset
+        size: _root.header.hash.size
+      signature_body:
+        pos: _root.header.signature.offset
+        size: _root.header.signature.size
+  vbmeta_auxiliary_data:
+    instances:
+      public_key_body:
+        pos: _root.header.public_key.offset
+        size: _root.header.public_key.size
+        type: vbmeta_public_key_header
+      public_key_metadata_body:
+        pos: _root.header.public_key_metadata.offset
+        size: _root.header.public_key_metadata.size
+      descriptors_body:
+        pos: _root.header.descriptors.offset
+        size: _root.header.descriptors.size
+        type: vbmeta_descriptors
+
+enums:
+  algorithm_types:
+    0: none
+    1: sha256_rsa2048
+    2: sha256_rsa4096
+    3: sha256_rsa8192
+    4: sha512_rsa2048
+    5: sha512_rsa4096
+    6: sha512_rsa8192
+  descriptor_types:
+    0: avb_descriptor_tag_property
```

Comment:
the prefix looks unneeded

Reply:
They are taken from this enum https://android.googlesource.com/platform/external/avb/+/refs/tags/android-11.0.0_r31/libavb/avb_descriptor.h#46